Improve outbound traffic performance.

thallgren opened the pull request:
## Description

The root-daemon now communicates directly with the traffic-manager
instead of routing all outbound traffic through the user-daemon. The
root-daemon uses a patched kubeconfig where `exec` configurations to
obtain credentials are dispatched to the user-daemon. This to ensure
that all authentication plugins will execute by the user and not by root. The old
behavior or routing everything through the user-daemon can be restored
by setting `cluster.connectFromUserDaemon` to `true` in `config.yml`

## Checklist

 - [x] I made sure to update `./CHANGELOG.yml`.
 - [ ] I made sure to add any docs changes required for my change (including release notes).
 - [x] My change is adequately tested.
 - [ ] I updated `DEVELOPING.md` with any special dev tricks I had to use to work on this code efficiently.
 - [ ] I updated `TELEMETRY.md` if I added, changed, or removed a metric name.
 - [ ] Once my PR is ready to have integration tests ran, I posted the PR in #telepresence-dev in the datawire-oss slack so that the "ok to test" label can be applied.

Commit:
Generalize the kubeconfig patcher and gRPC service.

This commit moves some code from the docker package to the patcher
package and to the k8s_config.go file, so that it's clear that the code
isn't docker specific.

The `authenticator` application is also removed in favor of using a
`telepresence kubeauth` subcommand. This, so that the installation using
one single binary will still work when kubeconfig exec patching is
necessary.

Signed-off-by: Thomas Hallgren <thomas@datawire.io>

Commit:
Improve outbound traffic performance.

The root-daemon now communicates directly with the traffic-manager
instead of routing all outbound traffic through the user-daemon. The
root-daemon uses a patched kubeconfig where `exec<` configurations to
obtain credentials are dispatched to the user-daemon. This to ensure
that all authentication plugins will execute in user-space. The old
behavior of routing everything through the user-daemon can be restored
by setting `cluster.connectFromUserDaemon` to `true` in `config.yml`

Signed-off-by: Thomas Hallgren <thomas@datawire.io>

sarabrajsingh commented on CHANGELOG.yml:
```diff
@@ -40,6 +40,20 @@ items:
+          The root-daemon now communicates directly with the traffic-manager instead of routing all outbound traffic
+          through the user-daemon. The root-daemon uses a patched kubeconfig where <code>exec</code> configurations to
+          obtain credentials are dispatched to the user-daemon. This to ensure that all authentication plugins will
+          execute in user-space. The old behavior or routing everything through the user-daemon can be restored by
```
```suggestion
          execute in user-space. The old behavior of routing everything through the user-daemon can be restored by
```

Commit:
Fix Test_ConnectWithKubeconfigExec failure on Windows.

Binary must suffixed with ".exe", because windows.

Signed-off-by: Thomas Hallgren <thomas@datawire.io>

Commit:
Make kubeauth a proper telepresence sub-command.

Signed-off-by: Thomas Hallgren <thomas@datawire.io>

Commit:
Don't run kubeauth containerized daemon test on Linux/arm64.

Using a `rootless` docker installation effectively prevents running
images from accessing the localhost through host.docker.internal. A
containerized daemon must be able to do that in order to authenticate
a cluster using `exec` plugins.

Signed-off-by: Thomas Hallgren <thomas@datawire.io>
